virtualizarr/readers/zarr.py

```diff
@@ -0,0 +1,329 @@
+                order=array_metadata_dict.get("order", None),
+                compressor=array_metadata_dict.get("compressor", None),
+                filters=array_metadata_dict.get("filters", None),
+                zarr_format=array_metadata_dict.get("zarr_format", None),
+            )
+
+            # 3c. Use the knowledge of the store location, variable name, and the zarr format to deduce which directory / S3 prefix the chunks must live in.
+            # ToDo Replace fsspec w/ Zarr python for chunk size: https://github.com/zarr-developers/zarr-python/pull/2426
+
+            import fsspec
+
+            array_mapper = fsspec.get_mapper(filepath + "/" + var)
+
+            # 3d. List all the chunks in that directory using fsspec.ls(detail=True), as that should also return the nbytes of each chunk. Remember that chunks are allowed to be missing.
+
+            # 3e. The offset of each chunk is just 0 (ignoring sharding for now), and the length is the file size fsspec returned. The paths are just all the paths fsspec listed.
+
+            # probably trying to do too much in one big dict/list comprehension
+            # uses fsspec.ls on the array to get a list of dicts of info including chunk size
+            # filters out metadata to get only chunks
+            # uses fsspec.utils._unstrip_protocol utility to clean up path
+
+            # array path to use for all chunks
+            array_path = fsspec.utils._unstrip_protocol(
+                array_mapper.root, array_mapper.fs
+            )
+
+            array_chunk_sizes = {
+                val["name"].split("/")[-1]: {
+                    "path": array_path,
+                    "offset": 0,
+                    "length": val["size"],
+                }
+                for val in array_mapper.fs.ls(array_mapper.root, detail=True)
+                if not val["name"].endswith((".zarray", ".zattrs", ".zgroup"))
+            }
+
+            # 3f. Parse the path and length information returned by fsspec into the structure that we can pass to ChunkManifest.__init__
+            # Initialize array chunk manifest from dictionary
+
+            array_chunkmanifest = ChunkManifest(array_chunk_sizes)
+
+            # 3g. Create a ManifestArray from our ChunkManifest and ZArray
+            array_manifest_array = ManifestArray(
+                zarray=array_zarray, chunkmanifest=array_chunkmanifest
+            )
+
+            # 3h. Wrap that ManifestArray in an xarray.Variable, using the dims and attrs we read before
+            array_variable = Variable(
+                dims=array_dims,
+                data=array_manifest_array,
+                attrs=array_metadata_dict.get("attributes", {}),
+                encoding=array_encoding,
+            )
+
+            virtual_variable_mapping[f"{var}"] = array_variable
+
+            all_array_dims.extend([dim for dim in array_dims])
+
+        coord_names = list(set(all_array_dims))
```

Comment:
This seems incorrect? I also don't actually see why you need to collect `all_array_dims`